morebits.js

```diff
@@ -2471,6 +2471,25 @@ Morebits.wiki.api.prototype = {
 
 };
 
+/** Retrieves wikitext from a page. Caching enabled, duration 1 day. */
+Morebits.wiki.getCachedWikitext = function(title, description, parent, callback) {
```

Comment:
Can return a promise here instead of using callbacks (Mb.w.api#post returns a promise). No need to follow the design of Morebits.wiki.* much of which is really old-fashioned. 

Also, do we need the `description`? I could be mistaken, but I think Morebits.status.init() isn't done until the form is submitted so the `description` is never actually displayed on the UI.

Reply:
Also, maybe it's better to call this getCachedJson and do the JSON.parse as well in this.

Reply:
As requested, I eliminated the description, and I renamed the function to `getCachedJson()`.

Regarding your feedback to use promises, are promises ES5? I thought they were ES6.

I began refactoring to use promises but I am having trouble passing the JSON variable from the `getCachedJson()` function to the `then()` function. I submitted a commit below with my attempt so far. Any tips?

Reply:
The native `Promise` is ES6, but jQuery has its own promises (which existed before promises were standardized as part of ECMAScript). Most MediaWiki code use jQuery promises (for instance, all the mw.Api methods return jQuery promises).

As to why it's not working, see the [jsdoc for Morebits.wiki.api](https://github.com/wikimedia-gadgets/twinkle/blob/277d5b6b52c/morebits.js#L2265). It'll work on changing 
```js
return new Morebits.wiki.api(description, query, function() {...}).post();
```
to
```js
return new Morebits.wiki.api(description, query).post().then(function() {...})